core/trino-main/src/main/java/io/trino/operator/PagesIndex.java

```diff
@@ -466,22 +463,8 @@ public PagesHashStrategy createPagesHashStrategy(List<Integer> joinChannels)
 
     private PagesHashStrategy createPagesHashStrategy(List<Integer> joinChannels, Optional<List<Integer>> outputChannels)
     {
-        try {
-            return joinCompiler.compilePagesHashStrategyFactory(types, joinChannels, outputChannels)
-                    .createPagesHashStrategy(ImmutableList.copyOf(channels));
-        }
-        catch (Exception e) {
-            log.error(e, "Lookup source compile failed for types=%s error=%s", types, e);
-        }
-
-        // if compilation fails, use interpreter
```

Comment:
Removing this path seems like it will break queries where the number of join columns exceeds the JVM class or method size limit (that currently rely on falling back to `SimplePagesHashStrategy`). The other removal site makes sense so long as the generated `PagesHashStrategy` for empty join channels is equivalent, but this one I'm less sure about. `SimplePagesHashStrategy` isn't especially complex, so I'm not sure I see an argument for this being a necessary simplification.